sklearn/utils/estimator_checks.py

```diff
@@ -287,7 +323,27 @@ def _construct_instance(Estimator):
     return estimator
 
 
-def check_estimator(Estimator):
+def _generate_instance_checks(name, estimator):
+    """Generate instance checks."""
+    yield from ((estimator, partial(check, name))
+                for check in _yield_all_checks(name, estimator))
+
+
+def _generate_class_checks(Estimator):
```

Comment:
can these helpers be used in `check_estimator`? Logic seems duplicted in various places